### ISSUE TITLE
Support electron 4+

### ISSUE THREAD

**Issue by franzwilhelm** [AlCalzone/node-tradfri-client#274]
Currently it's impossible to run the `node-tradfri-client` with an electron version newer than 3.x.x. I did a deep dig into why this is and found this: https://github.com/electron/electron/issues/16195

It turns out the `crypto` package bundled with newer versions of electron generates the following output:
```
var crypto = require('crypto')
const hashes = crypto.getHashes();
console.log(hashes);

[ 'md4', 'md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512' ]
```
This means we cannot expect `aes-128-ccm` nor `aes-128-gcm` to be included in the crypto package, even though we're running a modern node version. That means the code for authenticating with the tradfri package fails here: https://github.com/AlCalzone/node-dtls-client/blob/0520cf45b4bde9efda7b9cbaa79578355b4b0517/src/lib/AEADCrypto.ts#L48 since we meet the node 10+ requirement set here: https://github.com/AlCalzone/node-dtls-client/blob/0520cf45b4bde9efda7b9cbaa79578355b4b0517/src/lib/AEADCrypto.ts#L94

If the `node-aead-crypto` package were to be supported for node versions `> 10` if specified by a flag to the install script, it could function as a fallback for node use cases (like electron) where the aes ciphers are not available by default. This would require refactors of the install script in `node-aead-crypto` here: 
* https://github.com/AlCalzone/node-aead-crypto/blob/master/lib/install.js#L10 
* https://github.com/AlCalzone/node-aead-crypto/blob/master/package.json#L43

Although the solutions to this issue would mainly be resolved through the helper packages, I'm submitting the issue here. Please feel free to close it here, and I'll reopen in the respective repositories, or create linked issues if that helps. My current solution is to use pre-historic electron 3.x.x 👵

**Comment by AlCalzone** [comment]
This should be fixed with `node-aead-crypto@2.2.0`. That library now tries to natively execute the ciphers it polyfills. If any cipher fails, the module will be installed.
https://github.com/AlCalzone/node-aead-crypto/releases

**Comment by franzwilhelm** [comment]
@AlCalzone Cool! Will check this out later
